### PR TITLE
RDCC-1058: Remove @Ignore from Repository Tests

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1058

### Change description ###

Removed @Ignore and fixed repository unit tests in PRD

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/resources/db/migration/V1_1__init_tables.sql [hunk 1]
```diff
@@ -0,0 +1,142 @@
+-- NB Flyway requires lowercase for table names
```
Comment (AdilOoze) [inline src/test/resources/db/migration/V1_1__init_tables.sql]: These files have been added to **src/test/resources** (they are the same as those used in **src/main/resources**). This has been done so the database that is used in unit tests are built off these scripts and not those in **src/main** - as per discussions with @shashank-rastogi

Comment (shreedhar-hmcts) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: did you follow same in UP and Sync ? if not how did it work there ?

Comment (AdilOoze) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: Followed exact same in Sync, however in PRD we create two partial indexes which is supported in postgres but not yet supported by H2 so we cannot migrate our scripts via flyway due to this syntax deficiency in H2.

Comment (shreedhar-hmcts) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: Thats great finding 👍 . This justification will be helpful if some from QA ask us in future.

Comment (AdilOoze) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: I will add this to the **RDCC-1058 JIRA ticket** for future reference.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/repository/PrdEnumRepository.java [hunk 2]
```diff
@@ -11,4 +11,6 @@
 public interface PrdEnumRepository extends JpaRepository<PrdEnum, UUID> {
 
     List<PrdEnum> findByEnabled(String flag);
+
```
Comment (KOTLAPRA23) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/repository/PrdEnumRepository.java]: normally we should not write any new code only for test

File: src/test/resources/db/postgres/V6_1__adding_index_for_email_constraint.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+--CREATE UNIQUE INDEX deleted_not_null_idx ON professional_user (EMAIL_ADDRESS, DELETED)
+--WHERE DELETED IS NOT NULL;
+--
```
Comment (KOTLAPRA23) [inline src/test/resources/db/postgres/V6_1__adding_index_for_email_constraint.sql]: Please remove comment code from the file

Comment (AdilOoze) [reply src/test/resources/db/postgres/V6_1__adding_index_for_email_constraint.sql]: This has been addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/480/commits/dc4dc69e0df737dcc62fce07335d67d43f17c804

File: src/main/java/uk/gov/hmcts/reform/professionalapi/repository/PrdEnumRepository.java (hunk 2 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/repository/PrdEnumRepository.java]: This has been addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/480/commits/dc4dc69e0df737dcc62fce07335d67d43f17c804

File: src/test/resources/db/migration/V1_1__init_tables.sql (hunk 1 above)
Comment (KOTLAPRA23) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: remove all the scripts and create application.yml file inside test folder under  resources same as integration.
test/resources/application.yml

Comment (AdilOoze) [reply src/test/resources/db/migration/V1_1__init_tables.sql]: This has been addressed in this commit: https://github.com/hmcts/rd-professional-api/pull/480/commits/dc4dc69e0df737dcc62fce07335d67d43f17c804

File: src/test/java/uk/gov/hmcts/reform/professionalapi/helper/RepositorySetUp.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+import uk.gov.hmcts.reform.professionalapi.repository.OrganisationRepository;
+import uk.gov.hmcts.reform.professionalapi.repository.PaymentAccountRepository;
+import uk.gov.hmcts.reform.professionalapi.repository.PrdEnumRepository;
+import uk.gov.hmcts.reform.professionalapi.repository.ProfessionalUserRepository;
+import uk.gov.hmcts.reform.professionalapi.repository.UserAccountMapRepository;
+import uk.gov.hmcts.reform.professionalapi.repository.UserAttributeRepository;
+import uk.gov.hmcts.reform.professionalapi.util.RefDataUtil;
+
+public class RepositorySetUp {
+
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/helper/RepositorySetUp.java]: can you change the name to BaseRepository?

File: src/test/java/uk/gov/hmcts/reform/professionalapi/repository/OrganisationRepositoryTest.java [hunk 5]
```diff
@@ -47,12 +33,21 @@ public void test_findByOrganisationIdentifier() {
...
         assertThat(activeOrganisations).hasSize(1);
-        assertThat(activeOrganisations.get(0)).isEqualTo(activeOrg);
+        assertThat(activeOrganisations.get(0)).isEqualTo(organisation);
+    }
+
+    @Test
+    public void test_findByName() {
+        Organisation org = organisationRepository.findByName(organisation.getName());
+        assertThat(org).isEqualTo(organisation);
+    }
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/repository/OrganisationRepositoryTest.java]: have you written the custom equalsto() method to compare two objects of organisation?

**Pull request merged**